public/src/modules/translator.js

```diff
@@ -1,347 +1,462 @@
-;(function(translator) {
-	"use strict";
-	/* globals RELATIVE_PATH, config, define */
+/* global define, jQuery, config, RELATIVE_PATH, utils, window, Promise, winston */
 
-	var S = null;
-	var stringDefer = null;
+(function (factory) {
+	'use strict';
+	function loadClient(language, filename) {
+		return Promise.resolve(jQuery.getJSON(config.relative_path + '/language/' + language + '/' + (filename + '.json?v=' + config['cache-buster'])));
+	}
+	if (typeof define === 'function' && define.amd) {
+		// AMD. Register as a named module
+		define('translator', ['string'], function (string) {
+			return factory(string, loadClient);
+		});
+	} else if (typeof module === 'object' && module.exports) {
+		// Node
+		(function () {
+			require('promise-polyfill');
+			var fs = require('fs');
+			var path = require('path');
+			var winston = require('winston');
+			var plugins = require('../../../src/plugins');
+
+			function exists(filePath) {
+				return new Promise(function (resolve, reject) {
+					fs.stat(filePath, function (err, stats) {
+						if (err) {
+							if (err.code === 'ENOENT') {
+								return resolve(false);
+							}
+							return reject(err);
+						}
+						return resolve(stats.isFile());
+					});
+				});
+			}
 
-	// export the class if we are in a Node-like system.
-	if (typeof module === 'object' && module.exports === translator) {
-		exports = module.exports = translator;
-		S = require('string');
+			function readFile(filePath) {
+				return new Promise(function (resolve, reject) {
+					fs.readFile(filePath, {
+						encoding: 'utf-8'
+					}, function (err, data) {
+						if (err) {
+							reject(err);
+						} else {
+							resolve(data);
+						}
+					});
+				});
+			}
+
+			function loadServer(language, filename) {
+				var filePath = path.join(__dirname, '../../language', language, filename + '.json');
+				return exists(filePath).then(function (fileExists) {
+					if (!fileExists) {
+						if (plugins.customLanguageFallbacks[filename]) {
```

Comment:
This is no longer relevant as of bb80dc360eae252b85018eb99d63431b48641606, as `plugins.customLanguageFallbacks` is no longer set. If the target file is not found, assume `'{}'` instead (apostrophes included), since the data is added in `plugins.customLanguages` later on.

um... which got removed. See other note.